### DISCUSSION TITLE
atomWithStorage isn't initalizing localstorage with the initalValue, instead "null"

### DISCUSSION THREAD

**Discussion by spuckhafte** [pmndrs/jotai#2575] [Bug report]
```ts
export const anAtom = atomWithStorage("key", { ... });
```
In a file:
```ts
const [state, setState] = useAtom(anAtom);
...
```
Then I'm using this `state` in a react component. When it renders, an error pops that the `state` is not defined.
On checking the LocalStorage of the browser, it seems that the value of the `"key"` property is set to `null` (instead of the `initialValue` I set when creating the atom, i.e, `{ ... }`)

![image](https://github.com/pmndrs/jotai/assets/70335252/683c9470-9c30-48ad-b6a5-10a0454a1b74)

**Comment by dai-shi** [comment]
Can you please create a fresh reproduction with stackblitz?

**Comment by spuckhafte** [comment]
Very sorry, but it seems the error was from my side 🙏
